var/spack/repos/builtin/packages/papi/package.py

```diff
@@ -30,11 +31,57 @@ class Papi(Package):
     version('5.4.1', '9134a99219c79767a11463a76b0b01a2')
     version('5.3.0', '367961dd0ab426e5ae367c2713924ffb')
 
+    variant('example', default=True, description='Install the example files')
+    variant('cuda', default=False, description='Enable CUDA support')
+    variant('nvml', default=False, description='Enable NVML support')
+    variant('infiniband', default=False, description='Enable Infiniband support')
+    variant('powercap', default=False, description='Enable powercap interface support')
+    variant('rapl', default=False, description='Enable RAPL support')
+    variant('lmsensors', default=False, description='Enable lm_sensors support')
+
+    conflicts('+cuda', when='@:5.6.0')
+    conflicts('+nvml', when='@:5.6.0')
+
+    depends_on('cuda', when='+cuda')
+    depends_on('cuda', when='+nvml')
+    depends_on('lm-sensors', when='+lmsensors')
+
     # Does not build with newer versions of gcc, see
     # https://bitbucket.org/icl/papi/issues/46/cannot-compile-on-arch-linux
     patch('https://bitbucket.org/icl/papi/commits/53de184a162b8a7edff48fed01a15980664e15b1/raw', sha256='64c57b3ad4026255238cc495df6abfacc41de391a0af497c27d0ac819444a1f8', when='@5.4.0:5.6.99%gcc@8:')
 
+    def setup_environment(self, spack_env, run_env):
+        if '^cuda' in self.spec:
+            spack_env.set('CUDA_DIR', self.spec['cuda'].prefix)
+            run_env.prepend_path('LD_LIBRARY_PATH',
+                                 self.spec['cuda'].prefix.extras.CUPTI.lib64)
+
+            for path in self.spec['cuda'].libs.directories + self.rpath:
+                if path.find("/stubs") > -1:
+                    spack_env.remove_path("SPACK_RPATH_DIRS", path)
+                    run_env.remove_path("SPACK_RPATH_DIRS", path)
+
+            for path in self.spec['cuda'].libs.directories + [self.spec['cuda'].prefix.extras.CUPTI.lib64]:
```

Comment:
@G-Ragghianti flake is complaining about this line length (as well as line 67)

Reply:
Thanks.  Fixed the line lengths.